source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
> I think both of these lists shouldn't really exist and any restrictions should either be explicitly listed for each opcode or should be implied through capabilities.

This is a valid opinion, but again, it concerns the spec and not this code. We just follow the spec.

How about simply restricting the `spvOpcodeIsPointer()` call to only the Logical address model?  That can be accepted today, and it should save OpenCL implementations from improperly restrictive validation.